app/src/main/java/com/android/shelfLife/ui/recipes/GenerateRecipe.kt

```diff
@@ -0,0 +1,166 @@
+        },
+        label = { Text("Recipe Name") },
+        modifier = Modifier.fillMaxWidth())
+
+    // Error message for recipe name
+    recipeNameError?.let {
+      Text(
+          text = it,
+          color = androidx.compose.material.MaterialTheme.colors.error,
+          style = androidx.compose.material.MaterialTheme.typography.body2)
+    }
+
+    // Section for adding food items
+    Text(
+        text = "Food Items",
+        fontSize = 18.sp,
+        fontWeight = FontWeight.Bold,
+        modifier = Modifier.padding(vertical = 16.dp))
+
+    // Input field for a new food item
+    Row(verticalAlignment = Alignment.CenterVertically) {
+      OutlinedTextField(
+          value = newFoodItem,
+          onValueChange = {
+            newFoodItem = it
+            foodItemError = if (newFoodItem.isEmpty()) "Food item name cannot be empty" else null
+          },
+          label = { Text("Add Food Item") },
+          modifier = Modifier.weight(1f))
+
+      Button(
+          onClick = {
+            if (newFoodItem.isNotBlank()) {
+              foodItems.add(newFoodItem.trim())
+              newFoodItem = "" // Reset the input field
+            } else {
+              foodItemError = "Please enter a food item name"
+            }
+          },
+          modifier = Modifier.padding(start = 8.dp)) {
+            Text("Add")
+          }
+    }
+
+    // Error message for food item
+    foodItemError?.let {
+      Text(
+          text = it,
+          color = androidx.compose.material.MaterialTheme.colors.error,
+          style = androidx.compose.material.MaterialTheme.typography.body2)
+    }
+
+    // Display the list of food items
+    foodItems.forEachIndexed { index, foodItem ->
+      Row(
+          verticalAlignment = Alignment.CenterVertically,
+          modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp)) {
+            Text(text = "${index + 1}. $foodItem", modifier = Modifier.weight(1f), fontSize = 16.sp)
+
+            Button(onClick = { foodItems.removeAt(index) }) { Text("Remove") }
+          }
+    }
+
+    // Generate button
+    Button(
+        onClick = {
+          if (recipeName.isNotBlank() && foodItems.isNotEmpty()) {
+            val testIngredients =
+                foodItems.mapIndexed { index, name ->
+                  FoodItem(
+                      uid = index.toString(), foodFacts = FoodFacts(name, quantity = Quantity(1.0)))
+                }
+            navigationActions.goBack()
+            generationViewModel.updateRecipePrompt(
+                RecipePrompt(
+                    name = recipeName,
+                    ingredients = testIngredients,
+                    recipeType = RecipeType.HIGH_PROTEIN))
+            generationViewModel.generateRecipe(
+                onSuccess = { recipe ->
+                  generationViewModel.acceptGeneratedRecipe {
+                    Handler(Looper.getMainLooper()).post {
```

Comment:
Toats could be done directly in the UI to avoid Loopers and Handlers (again refer to AddFoodItem and EditFoodItem for inspiration)